components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,192 @@
+const axios = require("axios");
+const get = require("lodash/get");
+const moment = require("moment");
+const retry = require("async-retry");
+
+module.exports = {
+  type: "app",
+  app: "mailgun",
+  propDefinitions: {
+    domain: {
+      type: "string",
+      label: "domain",
+      description: "Domain used to send and receive email.",
+      async options(context) {
+        const q = context.query;
+        const options = [];
+        const results = await this.getAllDomains();
+        for (const domains of results) {
+          options.push(domains.name);
+        }
+        return options;
+      },
+    },
+    baseRegion: {
+      type: "string",
+      label: "Base region",
+      description:
+        'The region that Mailgun will use to send and receive email, which determines the URL to use when requesting Mailgun API. This must be the same region where "domain" was created.',
+      default: "US",
+      options: ["US", "EU"],
```

Comment:
Are these the only regions Mailgun supports?

Reply:
Yes, they only support two regions. Each region has its own API host, which is reflected in the code.